DOC-1969: lists and advlists plugins, text_patterns, and implicit and explicit list-creation UIs

brian-forte opened the pull request:
Ticket: lists and advlists plugins, text_patterns, and implicit and explicit list-creation UIs.

Changes:
* Removed references to the `lists` plugin being required for some text-patterns to work: the plugin is not so required.
* edits and additions to multiple files documenting the implicit and explicit list-creation UIs.
* Configuration example edits and additions.


Pre-checks:
- [x] Branch prefixed with `feature/6/` or `hotfix/6/`
- [x] Changelog entry added
- [x] Files has been included where required (if applicable)

Review:
- [ ] Documentation Team Lead has reviewed

brian-forte committed:
DOC-1969: text_patterns.adoc

1. Removed references to the `lists` plugin being required for some text-patterns to work: the plugin is not so required.

2. Edited the opening paragraphs for presentation clarity.

3. Edited, and altered the presentation of the notes documenting the use of each configuration example.

brian-forte committed:
DOC-1969.adoc

Added notes and example configurations documenting the explicit *and* implicit list-creation UIs now included in TinyMCE by default.

brian-forte committed:
DOC-1969: migration-from-5x.adoc

Added sub-section to Text Patterns section noting it being the cause of TinyMCE now coming with explicit and implicit list-creation UIs by default.

Also added xref pointing to lists.adoc and the documentation of these UIs therin.

brian-forte committed:
DOC-1969: text_patterns-and-lists.adoc

New partial file: /partials/configuration/text_patterns-and-lists.adoc.

Section added to new partial file documenting the presence of both implicit and explicit list creation UI.

brian-forte committed:
DOC-1969: admon-turning-text_patterns-off.adoc

New partial file: /partials/misc/admon-turning-text_patterns-off.adoc.

NOTE admonition added to new partial regarding the supported (and not supported) ways of turning `text_patterns` off.

This NOTE was already in `migration-from-5x.adoc`. Turning the note into a partial enables adding the same admonition to `lists.adoc`.

brian-forte committed:
DOC-1969: text_patterns.adoc

1. Copy-edits.

2. Comment added to block patterns example configuration re Advanced List Premium plugin requirement.

3. Added include statement pointing to `/configuration/text_patterns-and-lists.adoc`.

brian-forte committed:
DOC_1969: migration-from-5x.adoc

Replaced a section and NOTE admonition with include statements pointing to the same material.

brian-forte committed:
DOC-1969: lists.adoc

1. Copy edits.

2. expanded notes re the relationship between lists and text_patterns to note the relationship extends to Advanced Lists if that plugin is available.

3. include statement pointing to `misc/admon-turning-text_patterns-off.adoc` added.

4. comment re Advanced Lists Premium plugin requirement added to example configuration.

brian-forte committed:
DOC-1969: advlist.adoc

1.`toolbar: ‘bullist numlist’,` added  to basic setup example configuration.

2. include statement pointing to configuration/text_patterns-and-lists.adoc added.

brian-forte committed:
DOC-1969: changelog.md

Added entry for this ticket.

kemister85 commented on modules/ROOT/partials/configuration/text_patterns.adoc:
```diff
@@ -121,7 +158,7 @@ Replacement patterns must have the following:
 Whether a replacement pattern inserts a block or inline element depends on what the `+replacement+` string is.
 
-NOTE: Replacement patterns are executed on either pressing the *spacebar* or the *Enter* key.
+NOTE: Replacement patterns are executed on either pressing the *spacebar* or pressing the *ReTurn* or *Enter* key.
```
```suggestion
NOTE: Replacement patterns are executed on either pressing the *spacebar* or pressing the *Return* or *Enter* key.
```

kemister85 commented on modules/ROOT/pages/lists.adoc:
```diff
@@ -5,21 +5,76 @@
+
+Adding the `lists` plugin and the `bullist` and `numlist` toolbar items to a {productname} configuration enables the explicit list-creation user-interface (UI).
+
+The implicit list-creation UI comes via xref:content-behavior-options.adoc#text_patterns[text patterns]. The default set of `text_patterns` enabled when {productname} is initialised include https://daringfireball.net/projects/markdown/syntax[Markdown]-based patterns that {productname} automatically turns into basic ordered and unordered lists.
```
```suggestion
The implicit list-creation UI comes via xref:content-behavior-options.adoc#text_patterns[text patterns]. The default set of `text_patterns` enabled when {productname} is initialized include https://daringfireball.net/projects/markdown/syntax[Markdown]-based patterns that {productname} automatically turns into basic ordered and unordered lists.
```

kemister85 commented on modules/ROOT/pages/lists.adoc:
```diff
@@ -5,21 +5,76 @@
   plugins: 'lists',
-  toolbar: 'numlist bullist'
+  toolbar: 'bullist numlist',
+  `text_patterns` off.
```
Would be be best to change the below to `false` rather than `off`?
```suggestion
  `text_patterns` false,
```

brian-forte committed:
typo correction: s/ReTurn/Return/.

Co-authored-by: Karl Kemister-Sheppard <karlkemistersheppard@gmail.com>

brian-forte committed:
correction in configuration example: s/off/false/.

Co-authored-by: Karl Kemister-Sheppard <karlkemistersheppard@gmail.com>

brian-forte committed:
Spelling correction for en-US: s/initialised/initialized/.

Co-authored-by: Karl Kemister-Sheppard <karlkemistersheppard@gmail.com>

brian-forte committed:
DOC-1969: lists.adoc

1. error correction: advlist is not a Premium plugin.

2. Added further example: a complete, and specified, implicit lists UI without an explicit UI.

brian-forte committed:
DOC-1969: text_patterns.adoc

Error correction: advlist is not a Premium plugin.

MitchCTiny commented on modules/ROOT/pages/lists.adoc:
```diff
@@ -5,21 +5,106 @@
+----
+
+
+== Basic lists UI: explicit lists UI but no implicit lists UI
```
Is there any explanation for what is meant by explicit and implicit UI? I'm not sure I understand

TheSpyder commented on modules/ROOT/pages/lists.adoc:
```diff
@@ -5,21 +5,106 @@
+The `+lists+` plugin also normalizes list behavior between browsers. Enable it if you have problems with consistency making lists.
 
-== Basic setup
+It is important to note that, by default, {productname} includes both implicit and explicit list-creation user-interfaces.
+
+Adding the `lists` plugin and the `bullist` and `numlist` toolbar items to a {productname} configuration enables the explicit list-creation user-interface (UI).
+
+The implicit list-creation UI comes via xref:content-behavior-options.adoc#text_patterns[text patterns]. The default set of `text_patterns` enabled when {productname} is initialized include https://daringfireball.net/projects/markdown/syntax[Markdown]-based patterns that {productname} automatically turns into basic ordered and unordered lists.
+
+If the xref:advlist.adoc[Advanced Lists] Premium plugin is part of a {productname} configuration, the default `text_patterns` set also includes patterns that provide an implicit UI for generating the advanced ordered and unordered lists this plugin makes possible.
```
While I'm not opposed to calling it an `implicit ui`, I think we can still make this easier to follow.
```suggestion
It is important to note that, by default, {productname} includes both implicit and explicit list-creation user-interfaces.

Adding the `lists` plugin and the `bullist` and `numlist` toolbar items to a {productname} configuration enables the explicit list-creation user-interface (UI).

The implicit list-creation UI comes via xref:content-behavior-options.adoc#text_patterns[text patterns], which are enabled by default in {productname} version 6. The default set of `text_patterns` enabled when {productname} is initialized include https://daringfireball.net/projects/markdown/syntax[Markdown]-based patterns that {productname} automatically create ordered and unordered lists.
```

Following that, I agree the examples should be simplified and moved to the `text_patterns` documentation. I think we can lose the first two examples and just demonstrate that with the list plugin and list `text_patterns` config but no toolbar buttons, lists can still be inserted.

brian-forte committed:
DOC-1969: lists.adoc

Typo correction in code sample
